### ISSUE TITLE
serveStatic is non-intuitive

### ISSUE THREAD

**Issue by CodersBrothers** [restify/node-restify#535]
KISS: Keep it simple stupid. Please make it this for static files plugins, all people have the same problem, in my ofice 5 persons give me the same as bug!!

if you assign this:

``` javascript
server.get(/\/docs\/public\/?.*/, restify.serveStatic({
  directory: './public'
}));
```

will mean that if you get the url "domain.com/docs/public" this will be load the content of "public" folder. END. Not at prefix for the entire route path... and your fairy tales... Kiss men, kiss...

**Comment by phanimahesh** [comment]
The existing implementation of `serveStatic` is broken and the example straight out of docs does not work. I have wasted a couple of days due to that, and now switched to using module `serve-static`.

**Comment by micahr** [comment]
I agree with most on this thread. Serving static files isn't part of Restify's core competencies. Rather than continue to include a broken static file serving implementation, the existing plugin should be removed, in favor of pointing users to documentation on how to perform the same function. 

I'm in favor of spinning the existing plugin out to it's own module and allowing people to include it if they'd like.

**Comment by Maxwell2022** [comment]
What @micahr says ^^ +1

**Comment by shinymayhem** [comment]
`GET`ting static files is the epitome of REST! HTML is a media type that uses HATEOAS, so it is the original (true) REST API format.
I would love to see restify support this in an easy to use and intuitive way.

**Comment by wasa4587** [comment]
4 years after and still failing :/

**Comment by wasa4587** [comment]
In case someone have this issue, This worked for me :/

```
server.get(/\/?.(js|css|png|jpg)/, restify.plugins.serveStatic({
    directory: './public',
    default: '/index.html'
}));

server.get(/\/?.*/, restify.plugins.serveStatic({
    directory: './public',
    file: '/index.html'
}));
```

**Comment by yussan** [comment]
you can get documentation here http://restify.com/docs/plugins-api/#servestatic

**Comment by NicolaOrritos** [comment]
Had me lose 2 hours and still it's not working.  
Who mantains this project really doesn't care at all about this issue.  
Or isn't competent enough to fix it.

**Comment by yussan** [comment]
this is what i do on my website https://oopsreview.com , and it's worked until now

```
// serve static file from public directory
server.get(
  /\/?.*\//,
  restify.plugins.serveStatic({
    directory: `${__dirname}/../../public`,
    maxAge: 0
  })
)
```

ref: https://github.com/idmore/oopsreview-web/blob/master/src/server/index.js#L58
